### PR TITLE
Fix #3821: use numericInput for pricing

### LINKED ISSUE #3821: Prices should not be converted in admin view
### Expected behavior
The shop base currency should be honored for all prices shown in admin view.

### Actual behavior
If the shop's base currency is set to EUR and the admin does add a new product, the product's price get converted immediately from USD->EUR, because shop assumes that the entered value is USD.

EDIT: Strangely, the above stated does not happen reproducible

EDIT 2: Is happens reliably, if the currency selector in the nav bar doesn't match the shop base currency.

But the following seems to happen always:
Additionally it's not possible to enter the in the shop base locale, e.g. with a comma as separator for EUR currency.

### Steps to reproduce the behavior
- As an admin, set the shop base currency to EUR. And for good measure, also set the currency selector in the nav bar to EUR.
- Add a product
- Input a value
- The value gets converted as soon as the field blurs.

### Versions
Node: 8.2.1
NPM: 5.3.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.26.4
Reaction: 1.8.0
Reaction branch: release-1.8.0
Docker: 17.03.1-ce

### Analysis
It could have to do with this PR: https://github.com/reactioncommerce/reaction/commit/a9937023c911334d7cae5d6a84252666bd802b99

implicit_conversion

### PR DESCRIPTION
**Pull request by Akarshit**
Resolves #3821  
Impact: **major**  
Type: **bugfix**

## Issue
The component's value was changing because stale value was being stored in state.

## Solution
Used the `NumericInput` component modified by MJ for pricing.

## Testing
- As an admin, set the shop base currency to EUR.
- Add a product
- Input a value for price.
- The value **will not** get converted as soon as the field blurs.
- Publish the product, and observe that the price value was retained as is.

### CODE REVIEW

**Comment by Akarshit** [comment]
Looking into why the tests are failing.

**Comment by Akarshit** [comment]
Passing the test now :)

File: imports/plugins/included/default-theme/client/styles/numericInput.less [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+        > input {
+            width: 100%;
+            padding: 6px 12px;
+            border: 1px solid #cccccc;
+            border-radius: 2px;
+            background-color: #fafafa;
+            color: #505558;
+        }
+    }
+}
```
Comment (jshimko) [inline imports/plugins/included/default-theme/client/styles/numericInput.less]: Minor linting... this should all be 2 space indents.

**Comment by Akarshit** [comment]
@jshimko The error was 
1. You set your shop's currency as USD.
2. Then the currency rates are downloaded.
3. You set your shop'd currency as GBP. (but still rate of GBP is 0.71)
4. When the currency was being displayed in PDP, the rate was getting multiplied with the original value.
5. This was causing the values to be displayed incorrectly.

Fix ->
1. I have stopped currency conversion for admin view.

Also fixed the indenting error.

**Comment by jshimko** [comment]
After a fresh shop startup, I set my default currency as EUR and then tried editing the price of an existing product.  That's what ended up with the results I reported.  Is that expected behavior though?  At the very least, that's different than the current master branch behavior.

**Comment by Akarshit** [comment]
@jshimko Yes, that should be fixed now.

**Comment by jshimko** [comment]
I guess I don't see how my comment above isn't a bug.  

- fresh clone of this branch
- log in as admin
- set shop currency to EUR
- set current user currency to EUR
- create a new product
- set the price
- observe a percentage conversion being made on that price (presumably based on US dollars)

This doesn't happen on master and I don't see a way to set a price in EUR and _not_ have it converted.  Am I missing something?

**Comment by Akarshit** [comment]
@jshimko I tried the above steps but couldn't reproduce the bug after the last commit. Maybe I missed something checking again.

**Comment by Akarshit** [comment]
@jshimko Would it be best to pair on this, then maybe I would have better idea what the issue it.

**Comment by jshimko** [comment]
Yeah, sorry.  I think maybe I hadn't pulled the last commit.  I just did all of the testing again on the latest and I think we're good to go.  Thanks!

**Pull request merged**